src/diffusers/utils/callbacks.py

```diff
@@ -0,0 +1,116 @@
+from functools import partial
+from typing import Dict, List
+
+
+class PipelineCallback:
+    def __init__(self, **kwargs):
+        self.func = partial(self.callback_fn, **kwargs)
+
+    @property
+    def tensor_inputs(self) -> List:
+        raise NotImplementedError(f"You need to set the attribute `tensor_inputs` for {self.__class__}")
+
+    def callback_fn(self, pipeline, step_index, timesteps, callback_kwargs, **kwargs):
+        pass
+
+    def __call__(self, pipeline, step_index, timestep, callback_kwargs):
+        return self.func(pipeline, step_index, timestep, callback_kwargs)
+
+
+class MultiPipelineCallbacks:
+    def __init__(self, callbacks: List[PipelineCallback]):
+        self.callbacks = callbacks
+
+    @property
+    def tensor_inputs(self) -> List[str]:
+        return [input for callback in self.callbacks for input in callback.tensor_inputs]
+
+    def __call__(self, pipeline, step_index, timestep, callback_kwargs):
+        return {
```

Comment:
would something like this work?

        for callback in self.callbacks:
            callback_kwargs = callback(pipeline, step_index, timestep, callback_kwargs)

Reply:
yeah and it's easier to understand, I was with my optimize mentality